timelock-server/src/main/java/com/palantir/atlasdb/timelock/lock/AsyncLock.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.lock;
```

Comment:
My other worry would be exposing the LockService to products which then might mix usages of the 2 APIs expecting locks to work across APIs whereas they are actually distinct.  This is a case for tightening the API or keeping it internal to the transaction protocol.

There's also the Timestamp API which is unchanged. do we bump the version of that too?

(Sorry, this is just questions rather than answers!)